### PR TITLE
core/sched: sched.h: remove not needed bitarithm include to avoid conflict

### PR DESCRIPTION
**Pull request by kfessel**
bitarithm.h is not needed for the interface of shed but may cause conflicts
due to different definitions of SETBIT and CLRBIT

common implementations are: 

SETBIT(value, offset)  value |= (1<<offset) 
xor 
SETBIT(value, mask) value |= mask 

bitarithm implements the later

### CODE REVIEW

**Comment by maribu** [comment]
I'm not sure if there are indeed multiple definitions of `SETBIT()` in the code base, this will be sufficient. Maybe it is easier to rename the macro in the user code e.g. to `SETBIT_AT(value, offset)`, to be sure no issues arise when other headers actually needing `bitarithm` are included?

Anyway, dropping unneeded `#include`s is always a good idea. Let's run the CI to confirm this doesn't introduce any compilation issues.

**Comment by kaspar030** [comment]
Typo in commit message (shed->sched).

**Comment by kfessel** [comment]
typo fixed

**Comment by kaspar030** [comment]
Seems like CI found some occurences where the "bitarithm.h" is now missing...

**Comment by kfessel** [comment]
Am I able to reach the CI-report, or is it gone since i fixed the typo?

**Comment by kaspar030** [comment]
> Am I able to reach the CI-report, or is it gone since i fixed the typo?

yup, gone until the current build finishes.

**Comment by kfessel** [comment]
i would like to squash my last two commits which will fix the static-test failure detected by travis.

I am not sure about the cc1352 errors since this PR does neither touch it nor do i think it influences them.

**Comment by maribu** [comment]
Please squash the manual revert of the unrelated changes of the first commit into the first commit.

**Comment by benpicco** [comment]
The cc1352 build failures are addressed by #13282 - sorry about that.

File: cpu/sam0_common/periph/rtt.c [hunk 1]
```diff
@@ -32,7 +32,8 @@
  * effects, but simplifies the code. (This bit is always set on SAML21xxxxA)
  */
 #ifndef RTC_MODE0_CTRLA_COUNTSYNC
-#define RTC_MODE0_CTRLA_COUNTSYNC   BIT15
+#define RTC_MODE0_CTRLA_COUNTSYNC_Pos   15
+#define RTC_MODE0_CTRLA_COUNTSYNC       (0x1ul << RTC_MODE0_CTRLA_COUNTSYNC_Pos)
```
Comment (maribu) [inline cpu/sam0_common/periph/rtt.c]: Macro names in C should be all upper cases. (`POS` instead `Pos`)

File: drivers/at86rf2xx/at86rf2xx.c [hunk 2]
```diff
@@ -101,10 +101,7 @@ void at86rf2xx_reset(at86rf2xx_t *dev)
     }
 
     /* get an 8-byte unique ID to use as hardware address */
-    luid_get(addr_long.uint8, IEEE802154_LONG_ADDRESS_LEN);
-    /* make sure we mark the address as non-multicast and not globally unique */
-    addr_long.uint8[0] &= ~(0x01);
-    addr_long.uint8[0] |=  (0x02);
+    luid_get_eui64(addr_long);
```
Comment (maribu) [inline drivers/at86rf2xx/at86rf2xx.c]: Unrelated change

Comment (kfessel) [reply drivers/at86rf2xx/at86rf2xx.c]: Sorry

note to me: "never work in master"

File: cpu/sam0_common/periph/rtt.c [hunk 3]
```diff
@@ -32,7 +32,8 @@
  * effects, but simplifies the code. (This bit is always set on SAML21xxxxA)
  */
 #ifndef RTC_MODE0_CTRLA_COUNTSYNC
-#define RTC_MODE0_CTRLA_COUNTSYNC   BIT15
+#define RTC_MODE0_CTRLA_COUNTSYNC_Pos   15
```
Comment (maribu) [inline cpu/sam0_common/periph/rtt.c]:
```suggestion
#define RTC_MODE0_CTRLA_COUNTSYNC_POS   15
```

File: cpu/sam0_common/periph/rtt.c (hunk 1 above)
Comment (maribu) [inline cpu/sam0_common/periph/rtt.c]:
```suggestion
#define RTC_MODE0_CTRLA_COUNTSYNC       (0x1ul << RTC_MODE0_CTRLA_COUNTSYNC_POS)
```

**Comment by kfessel** [comment]
the mixed case _Pos is very common within at... register definitions, so I'd like to leave it this way or merge this the pos definition into the next line

Am 5. Februar 2020 09:04:49 MEZ schrieb Marian Buschsieweke <notifications@github.com>:

**Comment by maribu** [comment]
> the mixed case _Pos is very common within at... register definitions

RIOT coding convention is that preprocessor defines are all upper case. Other project are free to have their own coding conventions, but hey hardly matter to RIOT. Even if they are common.

**Comment by benpicco** [comment]
The defines are like that in the sam0 vendor files.
This is a fallback define if the define does not exist in the vendor file (yea, microchip is sometimes inconsistent), so it should be written just as the original define.

**Comment by maribu** [comment]
There is btw. such a define in the vendor files in `component/rtc.h`. And as the vendor files are externally supplied, they don't follow RIOT's coding convention. (So it is named `RTC_MODE0_CTRLA_COUNTSYNC_Pos`.) Why not just include this instead of defining this manually?

**Comment by maribu** [comment]
> This is a fallback define if the define does not exist in the vendor file (yea, microchip is sometimes inconsistent), so it should be written just as the original define.

OK, I see

**Comment by benpicco** [comment]
Please squash so `git bisect` still works.

**Comment by maribu** [comment]
> Please squash so `git bisect` still works.

If I understand you correctly, you want that compilation works for every random commit that git bisect might select.

This could also be reached by first adding the missing `#include`s (as including twice won't hurt due to include guards) and then removing the now unneeded include in the header. But I wouldn't mind to have both in a single commit.

**Comment by benpicco** [comment]
Right, you might also just reverse the order of commits.

**Comment by kaspar030** [comment]
> If I understand you correctly, you want that compilation works for every random commit that git bisect might select.

BTW, we at some point decided that commits within PR's don't have to compile individually.

**Comment by maribu** [comment]
> BTW, we at some point decided that commits within PR's don't have to compile individually.

Unless running Murdock on every commit merged in, this would also be hard to actually check...

Does someone know if git bisect actually runs through all commits by default? Most of the time I would want git to only consider merge commits (or commits directly applied to master in other projects that have these).

**Comment by kaspar030** [comment]
> Does someone know if git bisect actually runs through all commits by default?

Unfortunately it does, and I've not yet found a way to make it only consider merges. Also, some PR's have been merged as non-merge-commits (@aabadie, for dubious reasons? :wink:).

I usually do sth like ```make -Cfoo all -j8 || git bisect skip``` in each bisect step.

**Comment by kfessel** [comment]
i will squash: For the at86rf... define there are mutible sulutions
- since the bit is not uses for the non b rev i could just define it 0 (it is just uses in an OR so no harm done to the other Bits) and it should work an not throw an error (clean with a bit simplification)
- just use two different lines at the time of use and just not use the bit when it is not available
- merge the two lines and avoid to define the unused ..._Pos
- leave it as it is

I will squash using solution 4  and then you can decide

**Comment by maribu** [comment]
You can keep the `Pos`. The name was introduced by Microchip in the vendor files, so coding conventions don't apply

**Comment by miri64** [comment]
> I usually do sth like `make -Cfoo all -j8 || git bisect skip` in each bisect step.

Me as well. IIRC this just progresses to the next child (so not doing another binary search step).
